File: java/DataFrame/src/main/java/coyote/dataframe/U32Type.java
package coyote.dataframe;

import coyote.util.ByteUtil;

/** Type representing an unsigned, 32-bit value in the range of 0 to 4,294,967,295 */
public class U32Type implements FieldType
{
  private static final int _size = 4;

  private final static String _name = "U32";




  public boolean checkType( Object obj )
  {
    return obj instanceof Integer;
  }




  public Object decode( byte[] value )
  {
    return new Long( ByteUtil.retrieveUnsignedInt( value, 0 ) );
  }




  public byte[] encode( Object obj )
  {
    return ByteUtil.renderUnsignedInt( ( (Integer)obj ).intValue() );
  }




  public String getTypeName()
  {
    return _name;
  }




  public boolean isNumeric()
  {
    return true;
  }




  public int getSize()
  {
    return _size;
  }

}

File: csharp/DataFrame/Coyote/DataFrame/U32Type.cs
#region license
// Copyright (c) 2006 Stephan D. Cote' - All rights reserved.
// 
// This program and the accompanying materials are made available under the 
// terms of the MIT License which accompanies this distribution, and is 
// available at http://creativecommons.org/licenses/MIT/
// *
// Contributors:
//   Stephan D. Cote 
//      - Initial API and implementation
#endregion
using System;

namespace Coyote.DataFrame
{



    /// <summary>This is a type equivalent to the <code>uint</code> type, representing an unsigned, 32-bit value in the range of 0 to 4,294,967,295 </summary>
    public class U32Type : FieldType
    {
        private const int _size = 4;

        private const string _name = "U32";



        public virtual string TypeName
        {
            get
            {
                return _name;
            }
        }




        public virtual bool IsNumeric
        {
            get
            {
                return true;
            }
        }




        public virtual int Size
        {
            get
            {
                return _size;
            }
        }


        public virtual bool CheckType(object obj)
        {
            return (obj is uint);
        }




        public virtual object Decode(byte[] value)
        {
            return ByteUtil.RetrieveUnsignedInt(value, 0);
        }




        public virtual byte[] Encode(object obj)
        {
            return ByteUtil.RenderUnsignedInt((uint)obj);
        }







        public virtual string StringValue(byte[] val)
        {
            if (val == null)
            {
                return "";
            }
            else
            {
                object obj = Decode(val);
                if (obj != null)
                    return obj.ToString();
                else
                    return "";
            }
        }

    }

}